hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -511,6 +513,22 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		removeStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolUpdatingVersionConditionType)
 	}
 
+	// If reconciliation is paused we return before modifying any state
+	if isPaused, duration := supportutil.IsReconciliationPaused(log, nodePool.Spec.PausedUntil); isPaused {
+		md := machineDeployment(nodePool, controlPlaneNamespace)
+		err := pauseMachineDeployment(ctx, r.Client, md)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to pause MachineDeployment: %w", err)
+		}
+		ms := machineSet(nodePool, controlPlaneNamespace)
```

Comment:
Note that this covers the MS that is created when upgradeType is in place. But it does not cover the MachineSets created by the MachineDeployment when upgrade type is replace.
Which arguably should be handler by the MachineDeployment controller itself.

IMO this is good enough for now and we can iterate later on as this is improved upstream.
I'd just add a note in l518
```
TODO: Consider looking up and annotating the underlying MachineSets.

Also for paused to be complete we will need to pause all MHC if autorepair is enabled and remove the autoscaling labels from the MachineDeployment / Machineset. I would just add a TODO for this so we can follow up later on.